libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/CohortList.tsx

```diff
@@ -220,15 +220,29 @@ export class CohortList extends React.Component<
     const all = this.getAllCohort();
     if (index >= 0 && index < all.length) {
       const originCohort = all[index];
+      const duplicatedCohortNameStub =
+        originCohort.cohort.name + localization.Interpret.CohortBanner.copy;
+      let duplicatedCohortName = duplicatedCohortNameStub;
+      let cohortCopyIndex = 0;
+      while (this.existsCohort(all, duplicatedCohortName)) {
```

Comment:
I think this specific part, which resolves duplicates, can be some generic utility in core ui?  That way we can re-use the same logic for other name clashes.